src/psyclone/tests/psyir/transformations/inline_trans_test.py

```diff
@@ -1203,52 +1118,338 @@ def test_validate_named_arg(fortran_reader):
+    there is no use statement and the code for another routine with
+    the same name is specified within a module.
+
+    '''
+    code = f"{CALL_IN_SUB}{SUB_IN_MODULE}"
+    psyir = fortran_reader.psyir_from_source(code)
+    call = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    assert call.routine.is_unresolved
+    with pytest.raises(TransformationError) as info:
+        _ = inline_trans._find_routine(call)
+    assert ("Failed to find the source code of the unresolved routine "
+            "'inline_mod' after trying wildcard imports '[]' and all "
+            "routines that are not in containers." in str(info.value))
+
+
+def test_find_routine_unresolved_exception(fortran_reader):
+    '''Test that the expected exception is raised if the routine is
+    unresolved and can't be found in any wildcard imports or any 'raw'
+    subroutines.
+
+    '''
+    wildcard_use = CALL_IN_SUB_USE.replace(", only : sub", "")
+    code = (
+        f"{wildcard_use}"
+        f"module inline_mod\n"
+        f"end module inline_mod\n"
+        f"subroutine sub2()\n"
+        f"end subroutine sub2\n")
+    psyir = fortran_reader.psyir_from_source(code)
+    call = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    assert call.routine.is_unresolved
+    with pytest.raises(TransformationError) as info:
+        _ = inline_trans._find_routine(call)
+    assert ("Failed to find the source code of the unresolved routine 'sub2' "
+            "after trying wildcard imports '['inline_mod']' and all routines "
+            "that are not in containers." in str(info.value))
+
+
+def test_find_routine_import(fortran_reader):
+    '''Test that the routine can be found when there is a use statement
+    and the subroutine can be found in the associated module.
+
+    '''
+    code = f"{CALL_IN_SUB_USE}{SUB_IN_MODULE}"
+    psyir = fortran_reader.psyir_from_source(code)
+    call = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    assert call.routine.is_import
+    result = inline_trans._find_routine(call)
+    assert isinstance(result, Routine)
+    assert result.name == "sub"
+
+
+def test_find_routine_import_exception(fortran_reader):
+    '''Test that the routine raises the expected exception if the imported
+    module can't be found.
+
+    '''
+    code = f"{CALL_IN_SUB_USE}"
+    psyir = fortran_reader.psyir_from_source(code)
+    call = psyir.walk(Call)[0]
+    inline_trans = InlineTrans()
+    assert call.routine.is_import
+    with pytest.raises(TransformationError) as info:
+        _ = inline_trans._find_routine(call)
+    assert ("Failed to find the source for the imported routine 'sub' and "
+            "therefore cannot inline it." in str(info.value))
+
+
+def test_find_routine_module_to_raw_exception(fortran_reader):
+    '''Test that the routine raises the expected exception if the call
+    imports the routine and another routine with the same name is
```

Comment:
s/and/in the presence of a 'raw' subroutine of the same name/  or something. (Otherwise, I interpret this as meaning that we only raise the error if there is a local, raw subroutine of the same name and that's not right.)

Reply:
Reworded. Hopefully it is more clear.